contract/src/test/java/io/neow3j/contract/TransactionBuilderTest.java

```diff
@@ -84,137 +86,254 @@ public void setUp() {
+        b.signers(Signer.feeOnly(account1.getScriptHash()));
+        exceptionRule.expectMessage(new StringContains("fee-only witness scope"));
+        b.signers(Signer.feeOnly(account2.getScriptHash()));
+    }
+
+    // TODO: 14.09.20 Michael: Once TransactionAttributes are defined, write this test
+    @Ignore
+    @Test(expected = TransactionConfigurationException.class)
+    public void failAddingMoreThanMaxAttributesToTxBuilder() {
     }
 
     @Test
-    public void testCreationOfTheScript() throws IOException {
+    public void failWithoutSettingWallet() throws Throwable {
+        TransactionBuilder b = new TransactionBuilder(neow);
+        exceptionRule.expect(TransactionConfigurationException.class);
+        exceptionRule.expectMessage("wallet");
+        b.buildTransaction();
+    }
+
+    @Test
+    public void testAutomaticSettingOfValidUntilBlockVariable() throws Throwable {
         Wallet wallet = Wallet.createWallet();
         setUpWireMockForCall("invokescript", "invokescript_name_neo.json", SCRIPT);
+        ContractTestHelper.setUpWireMockForGetBlockCount(1000);
 
-        Invocation i = new Invocation.Builder(neow)
-                .withContract(NEO_TOKEN_SCRIPT_HASH)
-                .withFunction("name")
-                .withWallet(wallet)
-                .withValidUntilBlock(1000)
-                .build();
+        Transaction tx = new TransactionBuilder(neow)
+                .script(Numeric.hexStringToByteArray(SCRIPT))
+                .wallet(wallet)
+                .signers(Signer.feeOnly(wallet.getDefaultAccount().getScriptHash()))
+                .buildTransaction();
 
-        assertThat(Numeric.toHexStringNoPrefix(i.getTransaction().getScript()), is(SCRIPT));
+        assertThat(tx.getValidUntilBlock(),
+                is((long) NeoConstants.MAX_VALID_UNTIL_BLOCK_INCREMENT + 1000 - 1));
     }
 
     @Test
-    public void testAutomaticSettingOfSystemFee() throws IOException {
+    public void testAutomaticSettingOfSystemFee() throws Throwable {
         Wallet wallet = Wallet.createWallet();
         setUpWireMockForCall("invokescript", "invokescript_name_neo.json", SCRIPT);
-        Invocation i = new Invocation.Builder(neow)
-                .withScript(Numeric.hexStringToByteArray(SCRIPT))
-                .withWallet(wallet)
-                .withValidUntilBlock(1000)
-                .build();
-        assertThat(i.getTransaction().getSystemFee(), is(1007390L));
+        Transaction tx = new TransactionBuilder(neow)
+                .script(Numeric.hexStringToByteArray(SCRIPT))
+                .wallet(wallet)
+                .signers(Signer.feeOnly(wallet.getDefaultAccount().getScriptHash()))
+                .validUntilBlock(1000)
+                .buildTransaction();
+
+        assertThat(tx.getSystemFee(), is(1007390L));
 
         // With fail on false.
         String script = "10c00c046e616d650c14897720d8cd76f4f00abfa37c0edd889c208fde9b41627d5b5238";
```

Comment:
The second part of this test is not necessary anymore because we don't have this "fail on false" option anymore and it doesn't add anything to the test.

Reply:
True, I'll remove that part.